### ISSUE TITLE
More handlers - on closed connection, on closed channel

### ISSUE THREAD

**Issue by XVilka** [andersfugmann/amqp-client#26]
To be able properly handle automatic reconnect

**Comment by XVilka** [comment]
See example on how it is done in pika - https://pika.readthedocs.io/en/0.9.14/examples/asynchronous_consumer_example.html

**Comment by XVilka** [comment]
Fixed in https://github.com/andersfugmann/amqp-client/commit/795bbf530b6d59638771abc6e04af97092b826a0 
Thanks @andersfugmann !

**Comment by XVilka** [comment]
Well, closed too early. Still needs to be described in the examples, imho.

**Comment by andersfugmann** [comment]
I have implemented on_closed in a way that is backward compatible. If on_closed is not used, then an exception will be raised to the parent exception handler. An exception is of course not thrown if `on_closed` is called. 

The basic idea behind the library is to 'fail fast' idiom, so I prefer to have the application crash if any invariant is broken (i.e. a connection or channel is unexpectedly closed).  

I've also added an example.
